src/setup/node.rs

```diff
@@ -121,14 +147,22 @@ impl Node {
             .await
             .expect("Couldn't load the node's addresses.");
 
-        // TODO(Rqnsom) wait for the connection to confirm.
+        Node::wait_for_start(self.conf.net_addr.unwrap()).await;
     }
 
     /// Stops the node instance.
     pub fn stop(&mut self) -> io::Result<ChildExitCode> {
         // Cannot use 'mut self' due to the Drop impl.
 
+        // Remove address files since addresses may change if the node is restarted.
+        let remove_file = |file_name| match fs::remove_file(self.conf.path.join(file_name)) {
+            Ok(()) => (),
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => (),
+            Err(e) => panic!("Unexpected error: {:?}", e),
+        };
+        remove_file(NET_ADDR_FILE);
         self.conf.net_addr = None;
+        remove_file(REST_ADDR_FILE);
```

Comment:
You know how our test below does this:
```
Node->start()
node->stop()
// Start again:
Node->start().     // This part would fail after this change if we didn't implement this file deletion.
// Drop trait stops the node
```

Well, the thing is that the address file remained the same while the newly restarted node hadn't had time to update it. And our `Node` then decided to load in old addresses from the previous node run. Which meant it never managed to connect to it for the first time in the newly implemented `Node::wait_for_start`.